lib/matplotlib/dates.py

```diff
@@ -1905,6 +1911,47 @@ def axisinfo(self, unit, axis):
                               default_limits=(datemin, datemax))
 
 
-units.registry[np.datetime64] = DateConverter()
-units.registry[datetime.date] = DateConverter()
-units.registry[datetime.datetime] = DateConverter()
+# The following is so that we can set the converter for dates
+# via the validator for the rcParams `date.converter` and
+# `date.interval_multiples`
+
+class _rcParam_helper:
+    """
+    Never instatiated, but manages the rcParams for dates...
+    """
+
+    conv_st = 'auto'
+    int_mult = True
+
+    @classmethod
+    def set_converter(cls, s):
+        """Called by validator for rcParams date.converter"""
+        cls.conv_st = s
+        cls.register_converters()
+
+    @classmethod
+    def set_int_mult(cls, b):
+        """Called by validator for rcParams date.interval_multiples"""
+        cls.int_mult = b
+        cls.register_converters()
+
+    @classmethod
+    def register_converters(cls):
+        """
+        Helper to register the date converters when rcParams `date.converter`
+        and `date.interval_multiples` are changed.  Called by the helpers
+        above.
+        """
+        if cls.conv_st == 'concise':
+            converter = ConciseDateConverter
+        else:
+            converter = DateConverter
+
+        interval_multiples = cls.int_mult
+        print(interval_multiples)
+        units.registry[np.datetime64] = converter(
```

Comment:
perhaps just instantiate a single converter and assign to all 3?

Reply:
... I guess so?  the old code instantiated three converters, so I'm a little hesitant.  OTOH, I'm not sure why it did, and I don't think we lose any flexibility this way...